pkg/kubeclient/signup.go

```diff
@@ -77,27 +83,42 @@ func (c *userSignupClient) Update(obj *crtapi.UserSignup) (*crtapi.UserSignup, e
 	return result, nil
 }
 
-func (c *userSignupClient) ListByPhone(phone string) (*crtapi.UserSignupList, error) {
-	return c.listByHashedLabel(crtapi.BannedUserPhoneNumberHashLabelKey, phone)
-}
+// ListByPhoneNumberOrHash will return a list of UserSignups that have a phone number hash label value matching
+// the provided value.  If the value provided is an actual phone number, then the hash will be calculated and then
+// used to query the UserSignups, otherwise if the hash value has been provided, then that value will be used
+// directly for the query.
+func (c *userSignupClient) ListByPhoneNumberOrHash(phoneNumberOrHash string) (*crtapi.UserSignupList, error) {
+	if md5Matcher.Match([]byte(phoneNumberOrHash)) {
+		return c.listByLabel(crtapi.BannedUserPhoneNumberHashLabelKey, phoneNumberOrHash)
+	}
 
-// ListByHashedLabel returns a UserSignupList containing any UserSignup resources that have a label matching the specified label
-func (c *userSignupClient) listByHashedLabel(labelKey, labelValue string) (*crtapi.UserSignupList, error) {
+	// Default to searching for a hash of the specified value
+	return c.listByLabelForHashedValue(crtapi.BannedUserPhoneNumberHashLabelKey, phoneNumberOrHash)
+}
 
-	// Calculate the md5 hash for the phoneNumber
+// listByLabelForHashedValue returns a UserSignupList containing any UserSignup resources that have a label matching the
+// md5 hash of the specified value
+func (c *userSignupClient) listByLabelForHashedValue(labelKey, value string) (*crtapi.UserSignupList, error) {
```

Comment:
Same here. Something like `listNotDeactivatedSignupsByLabelForHashedValue()`

Reply:
Renamed to `listActiveSignupsByLabelForHashedValue()`